GliaWidgets/SecureConversations/SecureConversations.ChatWithTranscriptModel.swift

```diff
@@ -947,27 +946,56 @@ extension SecureConversations.TranscriptModel {
     }
 
     func receiveMessage(from messageSource: MessageSource) {
-        // We store message and try to determine if
-        // it has attachment. This depends on origin
-        // from which message was received. REST API
-        // does not return information about attachments,
-        // so we use messages delivered from socket instead.
         var list = receivedMessages[messageSource.id] ?? []
-        list.append(messageSource)
-        receivedMessages[messageSource.id] = list
-        let withPossibleAttachment = list.last(where: { $0.message.attachment != nil })?.message ?? messageSource.message
-        // We try to reuse outgoing message to show `delivered` status
-        // to indicated that message is delivered.
-        let outgoingMessage = list.last(where: { $0.outgoingMessage != nil })?.outgoingMessage ?? OutgoingMessage(
-            content: messageSource.message.content,
-            files: fileUploadListModel.succeededUploads.map(\.localFile)
-        )
-        self.replace(
-            outgoingMessage,
-            uploads: fileUploadListModel.succeededUploads,
-            with: withPossibleAttachment,
-            in: self.pendingSection
-        )
+
+        if list.isEmpty {
```

Comment:
I think it would be nice to clarify the logic behind empty list.

Reply:
I screwed up the fixup so I just rebased. The comment is here https://github.com/salemove/ios-sdk-widgets/pull/599/files#diff-43d1cf2ddce2070c280bb598aa0eb5d9fcf150bda3a89b61cc53035a048ea561R903